docs/content/en/docs/Coding Milestones/PoC2023q1/edge-syncer.md

```diff
@@ -0,0 +1,95 @@
+- EdgeSyncConfig (This is going to be replaced with EdgePlacement spec) 
+
+![edge-syncer deploy](./images/edge-syncer-deploy.png)
+
+After this, Edge-mc will put the following in the mailbox workspace.
+- Workload objects (both denatured one and not-denatured one)
+- API Binding (to API Export on the workload management workspace)
+- EdgeSyncConfig CR
+
+**TODO**: This is something we should clarify..e.g. which existing controller(s) in edge-mc will cover, or just create a new controller to handle uncovered one. 
+
+#### EdgeSyncConfig
+- The example is [here](https://gist.github.com/yuji-watanabe-jp/dfa9cdf35b06f612818446c2c3eb7969).
+- The CR here is used from edge syncer. 
+- The CR is placed at mb-ws to define
+  - object selector
+  - need of renaturing
+  - need of status upsync for downsyncing
+  - need of delete propagation for downsyncing
+  - Interval of downsyncing/upsyncing
+- The CR is managed by edge-mc
+  - At the initial implementation before edge-mc side controller become ready, we assume EdgeSyncConfig is on workload management workspace (wm-ws), and then which will be copied into mb-ws like other workload objects.
+  - This should be changed to be generated according to EdgePlacement spec. 
+- This CR is a placeholder for defining how edge-syncer behaves, and will be extended/splitted/merged according to further design discussion.
+
+#### Downsyncing
+
+- Edge syncer does downsyncing, which copy workload objects on mailbox workspace to p-cluster
+- If workload objects are deleted on mailbox workspace, the corresponding objects on the p-cluster will be also deleted according to EdgeSyncConfig. 
+- EdgeSyncConfig specifies which objects should be downsynced.
+  - object selector: group, version, kind, name, namespace (for namespaced objects), label, annotation
+- Cover cluster-scope objects and CRD
+  - CRD needs to be denatured if downsyncing is required.
+- Renaturing is applied if required (specified in EdgeSyncConfig).
+
+#### Renaturing
+- Edge syncer does renaturing, which converts workload objects to different forms of objects on a p-cluster. 
+- The conversion rules (downstream/upstream mapping) is specified in EdgeSyncConfig.
+- Some objects need to be denatured. 
+  - CRD needs to be denatured when conflicting with APIBinding.
+
+#### Status Upsyncing
+- Edge syncer does upsyncing status of downsynced objects at p-cluster to the status of objects on the mailbox workspace periodically. 
+  - TODO: Failing status upsync in some resources (e.g. deployment and service). Need more investigation. 
+- Status upsyncing on/off is configurable in EdgeSyncConfig. (default is on)
+
+#### Resource Upsyncing
+- Edge syncer does upsyncing resources at p-cluster to the corresponding mailbox workspace periodically. 
+- EdgeSyncConfig specifies which objects should be upsynced from p-cluster.
+  - object selector: group, version, kind, name, namespace (for namespaced objects), label, annotation (, and more such as ownership reference?)
```

Comment:
How much expressiveness is needed for C2P?  Would it suffice to identify the objects to upsync by API group, kind/resource, and namespace (for the ones that are namespaced)?  If not, what is the minimal additional restriction that is needed?

Reply:
I understand how that file works.  My question is whether something simpler and shorter would work.  For example, does something go wrong in the case of actual C2P usage if that file can not contain object names and instead syncing simply applies to all objects that meet the other criteria?

Reply:
e.g. secret should be downsycned selectively, not all.